app-modules/division/src/Observers/DivisionObserver.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Assist\Division\Observers;
+
+use Assist\Division\Models\Division;
+
+class DivisionObserver
+{
+    public function creating(Division $division): void
+    {
+        $division->createdBy()->associate($division->createdBy ?? auth()->user());
+        $division->updatedBy()->associate($division->updatedBy ?? auth()->user());
+    }
+
+    public function updating(Division $division): void
+    {
+        $division->updatedBy()->associate(auth()->user());
```

Comment:
Same as above with the `auth()->user()` check

Reply:
Disassociates as per our discussion.